lib/common/diagramvisitor.js

```diff
@@ -247,6 +260,20 @@ class DiagramVisitor {
             parameters.fileWriter.writeLine(0, `${source} ${DiagramVisitor.INHERITANCE} ${target}`);
         }
     }
+
+    /**
+     * Visitor design pattern
+     * @param {ClassDeclaration} classDeclaration - the object being visited
+     * @param {Object} parameters  - the parameter
+     * @protected
+     */
+    handleDecorators(classDeclaration, parameters) {
+        const decorators = [
+            ...classDeclaration.getDecorators(), // Class-level decorators
+            ...classDeclaration.getOwnProperties().flatMap(property => property.getDecorators()) // Property-level decorators
```

Comment:
Can we use the visitor pattern to do this tree navigation?

Reply:
I agree, I think this diverges from the pattern and not helping that much.

Reply:
Sure. Would update accordingly.